Scripts/Runtime/Entities/Lifecycle/AbstractEntityLifecycleStatusSystem.cs

```diff
@@ -0,0 +1,303 @@
+    {
+        private readonly List<EntityLifecycleStatus> m_EntityLifecycleStatus;
+        private readonly bool m_ShouldCleanupEntitiesCountAsDepartures;
+        private WorldEntityState m_WorldEntityState;
+        private NativeArray<JobHandle> m_Dependencies;
+        
+        /// <summary>
+        /// Creates a new <see cref="AbstractEntityLifecycleStatusSystem"/>
+        /// </summary>
+        /// <param name="shouldCleanupEntitiesCountAsDepartures">
+        /// If true, Entities that are in a CleanUp state with <see cref="ISystemStateComponentData"/>
+        /// will be considered as Departures. They will not show up later when they are actually destroyed by Unity.
+        /// </param>
+        protected AbstractEntityLifecycleStatusSystem(bool shouldCleanupEntitiesCountAsDepartures)
+        {
+            m_ShouldCleanupEntitiesCountAsDepartures = shouldCleanupEntitiesCountAsDepartures;
+            m_EntityLifecycleStatus = new List<EntityLifecycleStatus>();
+        }
+
+        protected override void OnCreate()
+        {
+            base.OnCreate();
+
+            m_WorldEntityState = WorldEntityState.GetOrCreate(World, this);
+
+            foreach (EntityLifecycleStatus entityLifecycleStatus in m_EntityLifecycleStatus)
+            {
+                entityLifecycleStatus.CreateQuery();
+            }
+
+            m_Dependencies = new NativeArray<JobHandle>(m_EntityLifecycleStatus.Count, Allocator.Persistent);
+        }
+
+        protected override void OnDestroy()
+        {
+            m_EntityLifecycleStatus.DisposeAllAndTryClear();
+            m_Dependencies.Dispose();
+
+            m_WorldEntityState.RemoveLifecycleStatusSystem(this);
+
+            base.OnDestroy();
+        }
+        
+        /// <summary>
+        /// Creates a new <see cref="IEntityLifecycleStatus"/> object to monitor Arrivals/Departures from a certain
+        /// archetype.
+        /// </summary>
+        /// <param name="componentTypes">The components to filter on.</param>
+        /// <returns>The <see cref="IEntityLifecycleStatus"/> object</returns>
+        protected IEntityLifecycleStatus CreateEntityLifecycleStatus(params ComponentType[] componentTypes)
+        {
+            EntityLifecycleStatus entityLifecycleStatus = new EntityLifecycleStatus(this, componentTypes);
+            m_EntityLifecycleStatus.Add(entityLifecycleStatus);
+            return entityLifecycleStatus;
+        }
+
+        protected override void OnUpdate()
+        {
+            JobHandle dependsOn = Dependency;
+            dependsOn = m_WorldEntityState.AcquireCreatedAndDestroyedEntities(
+                dependsOn,
+                out NativeArray<Entity>.ReadOnly createdEntities,
+                out NativeArray<Entity>.ReadOnly destroyedEntities,
+                m_ShouldCleanupEntitiesCountAsDepartures);
+
+            //If nothing was created or destroyed we can early exit
+            if (createdEntities.Length == 0
```

Comment:
Why do we care if `createdEntities.Length == 0` if we don't do anything with the collection?

Reply:
> we can early exit

😄

Reply:
But you don't use `createdEntities` at all afterward? Seems like you only care that `destroyedEntities.Length==0`

Reply:
Updated the comment to explain more what's going on. 

We don't use the `createdEntities` because it's not as efficient as using queries. But we do use the length as a signal to tell us that "new entities (of any type) have been created" therefore we want our LifecycleStatus objects to run their Updates which will use their underlying queries to detect if there are any new entities that match what they are looking for.

If I remove this, then they will only update and find new entities if we also happened to destroy entities too.